src/test/scala/com/eclipsesource/schema/SchemaValidatorSpec.scala

```diff
@@ -165,39 +167,37 @@ class SchemaValidatorSpec extends PlaySpecification {
       }
     }
 
-  "should resolve references on the classpath via UrlResolver" in {
-    val validator = SchemaValidator().addUrlResolver(ClasspathUrlResolver())
+  "should resolve references on the classpath with ClasspathUrlResolver" in {
+    val validator = SchemaValidator().addUrlResolver(ClasspathUrlProtocolHandler())
     // some.json references location.json within JAR
     val someJson = getClass.getResourceAsStream("/some.json")
     val schema = JsonSource.schemaFromStream(someJson)
     val result = validator.validate(schema.get, Json.obj("location" -> Json.obj("name" -> "Munich")))
     result.isSuccess must beTrue
   }
 
-  "should resolve references on the classpath via UrlResolver and custom protocols (#65)" in {
+  "should resolve relative references on the classpath with relative URL handler (#65)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
-      .shouldResolveRelativeRefsWithCustomProtocols(true)
-    // some.json references location.json within JAR
+      .addRelativeUrlHandler(new ClasspathHandler)
+    // some-issue-65.json references location.json within JAR
     val someJson = getClass.getResourceAsStream("/some-issue-65.json")
     val schema = JsonSource.schemaFromStream(someJson)
     val result = validator.validate(schema.get, Json.obj("location" -> Json.obj("name" -> "Munich")))
     result.isSuccess must beTrue
   }
 
-  "should resolve relative references with custom protocols (#65)" in {
+  "should resolve relative references on the classpath with relative URL handler (#65 - 2)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
-      .shouldResolveRelativeRefsWithCustomProtocols(true)
+      .addRelativeUrlHandler(new ClasspathHandler)
     val mySchemaJson = getClass.getResourceAsStream("/my-schema.schema")
     val mySchema = JsonSource.schemaFromStream(mySchemaJson)
     val result = validator.validate(mySchema.get, Json.obj("foo" -> Json.obj("bar" -> "whatever")))
     result.isSuccess must beTrue
   }
 
-  "should fail resolving relative references with custom protocols if option not set(#65)" in {
+  "should fail resolving relative references without any relative URL handler available (#65)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
+      .addUrlResolver(ClasspathUrlProtocolHandler())
```

Comment:
These tests are good, though I would make some recommendations to be exhaustive. You essentially have 2 options for the non-relative protocol handler and 3 options for the relative handler, so you should have 6 tests total:

1. No protocol handler, no relative handler
2. No protocol handler, classpath relative handler
3. No protocol handler, protocol-less classpath relative handler
4. Classpath protocol handler, no relative handler
5. Classpath protocol handler, classpath relative handler
6. Classpath protocol handler, protocol-less classpath relative handler

If you opened up the non-relative method to allow a general `URLStreamHandler` (so a protocol is not required), you would have 3 options for the non-relative handler and 3 for the relative handler, so 9 tests in total.